shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowDatabasesExecutor.java

```diff
@@ -54,18 +54,18 @@ public void execute(final BackendConnection backendConnection) {
     private Collection<Object> getSchemaNames(final BackendConnection backendConnection) {
         try {
             MetaDataContexts contexts = ProxyContext.getInstance().getMetaDataContexts();
-            SQLCheckEngine.check(new MySQLShowDatabasesStatement(), Collections.emptyList(), 
-                    contexts.getGlobalRuleMetaData().getRules(), backendConnection.getSchemaName(), contexts.getMetaDataMap(), backendConnection.getGrantee());
-            return new ArrayList<>(ProxyContext.getInstance().getAllSchemaNames());
+            SQLCheckEngine.check(new MySQLShowDatabasesStatement(), Collections.emptyList(), contexts.getGlobalRuleMetaData().getRules(), backendConnection.getSchemaName(), contexts.getMetaDataMap(),
+                    backendConnection.getGrantee());
         } catch (final SQLCheckException ex) {
-            Collection<Object> result = new LinkedList<>();
-            for (String each : ProxyContext.getInstance().getAllSchemaNames()) {
-                if (SQLCheckEngine.check(each, getRules(each), backendConnection.getGrantee())) {
-                    result.add(each);
-                }
+            // ignore
+        }
```

Comment:
Is there a need to do such modification?

Reply:
Hey @terrymanu 
@lwclover and I were confused here, actually it is not straightforward enough, could you have a look?

Reply:
Maybe cause problem if SQLCheckException occur, why ignore the exception here?

Reply:
I think codes below are redundant lines. 
```
MetaDataContexts contexts = ProxyContext.getInstance().getMetaDataContexts();
            SQLCheckEngine.check(new MySQLShowDatabasesStatement(), Collections.emptyList(), contexts.getGlobalRuleMetaData().getRules(), backendConnection.getSchemaName(), contexts.getMetaDataMap(),
                    backendConnection.getGrantee());
```